apps/emqx/src/emqx_tls_lib.erl

```diff
@@ -205,8 +214,20 @@ default_versions(_) ->
     lists:delete('tlsv1.3', proplists:get_value(available, ssl:versions())).
 
 %% Deduplicate a list without re-ordering the elements.
-dedup([]) -> [];
-dedup([H | T]) -> [H | dedup([I || I <- T, I =/= H])].
+dedup([]) ->
+    [];
+dedup(List0) ->
+    List = lists:foldl(
```

Comment:
Tiny nit: maybe `ordsets` might be more suitable for removing duplicates, instead of searching the whole list on each membership test.  Alternatively, maybe using `foldr` can avoid the `lists:reverse` at the end?

Reply:
foldr is not ok for this.
we must choose element from left. such as:
[1,2,2,3,2,4] ->  foldl -> [1,2,3,4]
                     -> foldr -> [1,3,2,4]

Reply:
`ordsets` is sorted by element. is not suitable for this too.
the `sets` is ok for this, but I see sets's source code.
https://github.com/erlang/otp/blob/master/lib/stdlib/src/sets.erl#L173-L189
it also uses lists:memeber/2 and constructs a complex structure to meet the other operations of set(such as union)